File: java/scon-core/src/test/java/io/github/realmlabs/scon/ParserParseFuzzTest.java
package io.github.realmlabs.scon;

import com.code_intelligence.jazzer.junit.FuzzTest;
import java.nio.charset.StandardCharsets;
import org.junit.jupiter.api.condition.EnabledIfSystemProperty;

@EnabledIfSystemProperty(named = "scon.fuzz.enabled", matches = "true")
final class ParserParseFuzzTest {
    @FuzzTest(maxDuration = "30s")
    void parse(byte[] data) {
        String source = new String(data, StandardCharsets.UTF_8);
        try {
            Scon.parseString(source);
        } catch (SconException ignored) {
            // Expected syntax and semantic errors are not fuzz failures.
        }
    }
}

File: kotlin/scon-core/src/test/kotlin/io/github/realmlabs/scon/ParserParseFuzzTest.kt
package io.github.realmlabs.scon

import com.code_intelligence.jazzer.junit.FuzzTest
import org.junit.jupiter.api.condition.EnabledIfSystemProperty

@EnabledIfSystemProperty(named = "scon.fuzz.enabled", matches = "true")
@EnabledIfSystemProperty(named = "scon.fuzz.target", matches = "parse|all")
class ParserParseFuzzTest {
    @FuzzTest(maxDuration = "30s")
    fun parse(data: ByteArray) {
        val source = data.toString(Charsets.UTF_8)
        runCatching { resolveSource(source) }
    }
}
